lntest/itest/lnd_test.go

```diff
@@ -5387,6 +5387,87 @@ func testInvoiceRoutingHints(net *lntest.NetworkHarness, t *harnessTest) {
 	cleanupForceClose(t, net, net.Alice, chanPointEve)
 }
 
+// testInvoiceCheckInboundBandwidth tests that creating a new invoice in a node
+// without sufficient inbound bandwidth fails when the check for it is enabled.
+//
+// The scenario in which there is enough bandwidth is also tested.
+func testInvoiceCheckInboundBandwidth(net *lntest.NetworkHarness,
```

Comment:
Instead of adding an integration test, check if it can be made a unit test on `AddInvoice` 😄

Reply:
This makes a lot of sense, thanks @halseth. I am checking the code and I am a bit confused so a bit of help would be very appreciated.

My changes are on `lnrpc/invoicesrpc` where there are not unit tests and as far as I understand the behaviour is covered with integration tests. I see how the unit test could in theory cover `channeldb`'s `AddInvoice` but I did not modify that part. I tried to do the check sooner than later so the db does not handle this.

Reply:
You can add a new file `lnrpc/invoicesrpc/addinvoice_test.go` for the unit test.

Reply:
@halseth I've been reviewen this again after a few days I am not sure how this could be a unit test because to check if there are enough inbound funds we would need to first create channels so it would require to touch channeldb.

On top of that I also found out that the contribution guide asks that anything that changes behaviour within in the RPC be accompanied by integration tests. So I am a bit lost here right now. Any hint on what I may be missing would be very very helpful.

Reply:
If you look at `AddInvoiceConfig` you can mock out several of the dependencies. Since the actual logic here is pretty self contained you could also create a method such as `checkInboundBandwidht(chans *OpenChannel, amt Millisatoshi) (bool, error)` and unit test that method directly.

Reply:
Small update: I moved the logic to `checkInboundBandwidth(...)` as you proposed and unit tested it.